modules/transport-netty4/src/test/java/org/elasticsearch/rest/Netty4HeadBodyIsEmptyIT.java

```diff
@@ -197,11 +198,15 @@ private void headTestCase(
             final String url,
             final Map<String, String> params,
             final int expectedStatusCode,
-            final Matcher<Integer> matcher) throws IOException {
+            final Matcher<Integer> matcher,
+            final String... expectedWarnings) throws IOException {
         Request request = new Request("HEAD", url);
         for (Map.Entry<String, String> param : params.entrySet()) {
             request.addParameter(param.getKey(), param.getValue());
         }
+        request.setOptions(expectVersionSpecificWarnings(w -> {
```

Comment:
In #36511 I added `expectWarnings`, which is syntactic sugar for this call. It might be nice to switch to that when it merges (should be very soon).